File: examples/transforms/approval_up.py
#!/usr/bin/env python3
import json, sys
x = json.load(sys.stdin)
approval = x.get("approval")
if approval in ("manual", "automatic"):
    print(json.dumps({"approved": True}))
elif approval == "rejected":
    print(json.dumps({"approved": False}))
else:
    print(f"unsupported approval: {approval}", file=sys.stderr)
    sys.exit(12)

File: examples/transforms/approval_up.js
#!/usr/bin/env node
"use strict";

const input = readJson();
const approval = input.approval;

if (approval === "manual" || approval === "automatic") {
  writeJson({ approved: true });
} else if (approval === "rejected") {
  writeJson({ approved: false });
} else {
  process.stderr.write(`unsupported approval: ${approval}\n`);
  process.exitCode = 12;
}

function readJson() {
  return JSON.parse(require("node:fs").readFileSync(0, "utf8"));
}

function writeJson(value) {
  process.stdout.write(`${JSON.stringify(value)}\n`);
}
